DANS-KNAW/easy-solr4files-index
# EASY-1325: search servlet

Fixes EASY-1325

#### When applied 
* the solr core won't produce `attr_` prefixes in indexed field names, though now again images and pakbon contents are not indexed, just their metadata (#8 embedded intis PR)
* filters queries on accessibility and data available
* apply optional basic authentication
* [x] filters on accessible-to unless the (optionally) authenticated user is the owner of the dataset or the user is an archivist or administrator

Next PR's
* rename the project to easy-file-index (and change easy-build accordingly)
* new issues to distill from the todo's in the source

#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?

```
mvn clean install && vagrant destroy -f && vagrant up
curl -X POST 'http://test.dans.knaw.nl:20150/fileindex/init'
curl -v -u user001:user001 'http://test.dans.knaw.nl:20150/filesearch?text=voor' | grep file_path
```

The `-v` option shows the response headers, the `-u` option is also optional.
`user001` and `easyadmin` will get two files, other users just one when searching for the word `voor`.
See the new readme for variations of both curl commands and command line variants of the first curl command.

The bag store on the local VM is initialized with two test bags which get indexed with the first curl command. It takes a while but you can load additional bags by copying the content of `/tmp/pdbs.tar` on `teasy.dans.knaw.nl` into `/data/bagstores/pdbs` of the local VM.

**Note / prerequisites for vagrant up**
I had the following directories avaiable in `~/.ansible/roles/`, synchronizing and running dtap might help

dans.capture-deployer-email
dans.easy-add-new-license
dans.easy-bag-index
dans.easy-bag-store
dans.easy-includes
dans.easy-ldap-dir
dans.easy-send-metadata-to-datacite
dans.easy-test-users
dans.local-test-vm-base
dans.local-yum-repo
dans.postgresql
dans.solr

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-update-solr4files-index    | #8     | accepted but ~~not yet merged~~ closed PR is included in this one [differences since then](https://github.com/DANS-KNAW-jp/easy-update-solr4files-index/pull/1/)
